refactor(breadcrumb): Upgrade Breadcrumb component from AntD v4 to AntD v5

EnxDev opened the pull request:
<!---
Please write the PR title following the conventions at https://www.conventionalcommits.org/en/v1.0.0/
Example:
fix(dashboard): load charts correctly
-->

### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
This PR migrates the Breadcrumb component from Ant Design v4 to v5, ensuring compatibility with the latest API changes while keeping the implementation as vanilla as possible.
- Upgraded Breadcrumb to use AntD v5.
- Created a story to CSF3 format for better maintainability and compatibility.
- Added a test file to validate the component behavior.
- Kept the implementation minimalistic and close to native AntD usage to reduce unnecessary complexity.

Furthermore, move existing stories to be grouped under the 'Components' section in Storybook and add a missing story for the FaveStar component to demonstrate its behavior

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

- Before Breadcrumb inside a DrillByModal
https://github.com/user-attachments/assets/17e35cf4-c01e-4174-9e50-0046630eee4b

- After Breadcrumb inside aDrillByModal
https://github.com/user-attachments/assets/2141f4eb-99b0-4ade-b975-fc824fecd784


- Before
![before-breadcrumb](https://github.com/user-attachments/assets/7d1c332e-a414-472e-8a31-633ace46bb3d)

- After
![after-breadcrumb](https://github.com/user-attachments/assets/c532fc05-305a-4869-8e3e-28b5452d6859)

- Storybook before
![storybook-before](https://github.com/user-attachments/assets/3180215c-019e-4cac-89f5-fb206170ca6e)

- Storybook after
![storybook-after](https://github.com/user-attachments/assets/34b06607-154e-4d4d-a3d0-707df1ba82ed)

### TESTING INSTRUCTIONS

<!--- Required! What steps can be taken to manually verify the changes? -->
- All tests should pass

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

EnxDev committed:
feat(Breadcrumb): add wrapper for Breadcrumb component, with test and story

geido commented on superset-frontend/src/components/Breadcrumb/Breadcrumb.stories.tsx:
```diff
@@ -0,0 +1,57 @@
+      defaultValue: '/',
+    },
+    items: {
+      control: 'object',
```
Let's use `false` here for the control as it is not really convenient to pass array of items in Storybook and let's use the Storybook table info and summary to provide an example for this prop.

geido commented on superset-frontend/src/components/Breadcrumb/Breadcrumb.test.tsx:
```diff
@@ -0,0 +1,37 @@
+      </Breadcrumb>,
+    );
+
+    expect(screen.getByText('Home')).toBeInTheDocument();
```
Can we test that also a _**custom**_ divider is present

msyavuz replied:
Agreed

geido commented on superset-frontend/src/components/Breadcrumb/Breadcrumb.stories.tsx:
```diff
@@ -0,0 +1,57 @@
+export default meta;
+type Story = StoryObj<typeof Breadcrumb>;
+
+export const Default: Story = {
```
I see that the component has a bunch of [different abilities](https://ant.design/components/breadcrumb), such as showing a dropdown. It would be nice to have a gallery here of these different abilities. Something like https://superset-storybook.netlify.app/?path=/story/buttongroup--interactive-button-group where we can see multiple components in one view.

msyavuz replied:
I agree with this as well

geido commented on superset-frontend/src/components/Breadcrumb/index.tsx:
```diff
@@ -0,0 +1,21 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Breadcrumb as antdBreadcrumb } from 'antd-v5';
```
```suggestion
import { Breadcrumb as AntdBreadcrumb } from 'antd-v5';
```

geido commented on superset-frontend/src/components/Breadcrumb/index.tsx:
```diff
@@ -0,0 +1,21 @@
+ */
+import { Breadcrumb as antdBreadcrumb } from 'antd-v5';
+
+export const Breadcrumb = antdBreadcrumb;
```
```suggestion
export const Breadcrumb = AntdBreadcrumb;
```

EnxDev committed:
Merge branch 'template_less' into enxdev/refactor/migrate-Breadcrumb-to-antd5

EnxDev committed:
refactor(Breadcrumb): ensure separators are correctly displayed in tests

EnxDev committed:
refactor Breadcrumb story

EnxDev committed:
Merge branch 'template_less' into enxdev/refactor/migrate-Breadcrumb-to-antd5

EnxDev committed:
chore(storybook): move stories under 'Components' and add FaveStar story
- Move existing stories to be grouped under the 'Components' section in Storybook
- Add a story for the FaveStar component to demonstrate its behavior

EnxDev committed:
refactor(Breadcrumb): improve story

EnxDev committed:
refactor(Breadcrumb): migrate component to antd5

EnxDev committed:
refactor(DrillByModal): remove custom Breadcrumb component

EnxDev committed:
Merge branch template_less into enxdev/refactor/migrate-Breadcrumb-to-antd5

EnxDev committed:
refactor(IconButton): enhance story

EnxDev committed:
lint fix

EnxDev committed:
fix CI test

EnxDev committed:
fix DrillByModal RTL test

EnxDev committed:
fix Breadcrumb items

EnxDev committed:
Merge branch template_less into enxdev/refactor/migrate-Breadcrumb-to-antd5

EnxDev committed:
fix type for breadcrumbItems

msyavuz commented on superset-frontend/src/components/Alert/Alert.stories.tsx:
```diff
@@ -30,7 +30,7 @@ const bigText =
 
 export default {
-  title: 'Alert',
+  title: 'Components/Alert',
```
I think it looks better this way too. But if we start doing this it should be standardized as well. cc @geido

EnxDev replied:
> I think it looks better this way too. But if we start doing this it should be standardized as well. cc @geido

For now, I don't think there's a clear convention. Previously, some components were inside the components folder and others outside, without really following a specific structure/convention. So I decided to stick with the same organization that already exists in the codebase for now.

@rusackas maybe we could align on a common structure together with the rest of the community?

msyavuz commented on superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx:
```diff
@@ -308,8 +311,35 @@ export default function DrillByModal({
   );
-
-  const breadcrumbs = useDrillByBreadcrumbs(breadcrumbsData, onBreadcrumbClick);
+  const breadcrumbItems = breadcrumbsData
```
I am a bit confused here. We used to have a hook where we extracted this logic into. That hook seems to be unused now? Maybe we should modify the hook to return items and pass those items to breadcrumb here?

EnxDev replied:
I think this could have the same answer as  [here](https://github.com/apache/superset/pull/32905#discussion_r2039285346), thanks!

msyavuz commented on superset-frontend/src/components/Chart/DrillBy/useDrillByBreadcrumbs.tsx:
```diff
@@ -26,23 +26,23 @@ import {
   styled,
```
This file is entirely unused if we move the logic back into the DrillByModal. I think having the logic here and using this hook in DrillByModal makes better sense.

EnxDev replied:
We were deciding whether to remove the hook or not, since after the refactor the logic has been simplified — there's less code and it's only used here. I'll remove it, thanks!

msyavuz commented on superset-frontend/src/components/index.ts:
```diff
@@ -24,8 +24,11 @@
 // eslint-disable-next-line no-restricted-imports
-export { Breadcrumb as AntdBreadcrumb } from 'antd';
-
+export {
+  Checkbox as AntdCheckbox,
+  Collapse as AntdCollapse,
+  Select as AntdSelect,
+} from 'antd';
```
I don't think we should add these imports again here. These component are all migrated to Antd v5.

EnxDev replied:
> I don't think we should add these imports again here. These component are all migrated to Antd v5.

I think is a regression after the merge, good catch!

msyavuz reviewed (changes requested):
If we handle the DrillByModal thing i think this is good to merge.

EnxDev committed:
remove unused useDrillByBreadcrumbs.tsx

EnxDev committed:
remove imports mistakenly added during merge causing regression

geido reviewed (approved):
Thanks!
